File: private-tools/addr.js
const bs58check = require('bs58check');
const bs58check = require('bs58check');
const bs58check = require('bs58check');
const bs58check = require('bs58check');
let { bech32, bech32m } = require('bech32')
let { bech32, bech32m } = require('bech32')
let { bech32, bech32m } = require('bech32')
let { bech32, bech32m } = require('bech32')




function addressToHex(address) {
function addressToHex(address) {
function addressToHex(address) {
function addressToHex(address) {
    let bytes;
    let bytes;
    let bytes;
    let bytes;
    if (address.startsWith('1') || address.startsWith('3')) {
    if (address.startsWith('1') || address.startsWith('3')) {
    if (address.startsWith('1') || address.startsWith('3')) {
    if (address.startsWith('1') || address.startsWith('3')) {
        // Base58Check address
        // Base58Check address
        // Base58Check address
        // Base58Check address
        bytes = bs58check.decode(address).toString('hex');
        bytes = bs58check.decode(address).toString('hex');
        bytes = bs58check.decode(address).toString('hex');
        bytes = bs58check.decode(address).toString('hex');
    } else if (address.startsWith('bc1')) {
    } else if (address.startsWith('bc1')) {
    } else if (address.startsWith('bc1')) {
    } else if (address.startsWith('bc1')) {
        // Bech32 address
        // Bech32 address
        // Bech32 address
        // Bech32 address
        const decoded = bech32.decode(address);
        const decoded = bech32.decode(address);
        const decoded = bech32.decode(address);
        const decoded = bech32.decode(address);
        console.log("prefix: ", decoded.prefix);
        console.log("prefix: ", decoded.prefix);
        console.log("prefix: ", decoded.prefix);
        console.log("prefix: ", decoded.prefix);
        const words = bech32.fromWords(decoded.words.slice(1)); // remove the version byte
        const words = bech32.fromWords(decoded.words.slice(1)); // remove the version byte
        const words = bech32.fromWords(decoded.words.slice(1)); // remove the version byte
        const words = bech32.fromWords(decoded.words.slice(1)); // remove the version byte
        return "0014" + Buffer.from(words).toString('hex');
        return "0014" + Buffer.from(words).toString('hex');
        return "0014" + Buffer.from(words).toString('hex');
        return "0014" + Buffer.from(words).toString('hex');
    } else {
    } else {
    } else {
    } else {
        throw new Error('Invalid Bitcoin address');
        throw new Error('Invalid Bitbi address');
        throw new Error('Invalid Bitcoin address');
        throw new Error('Invalid Bitcoin address');
    }
    }
    }
    }
    return bytes;
    return bytes;
    return bytes;
    return bytes;
}
}
}
}




const address = 'bc1qc2ky0nanf5vgtqc080fawm8ep9tqk4nykummrg';
const address = 'bc1qc2ky0nanf5vgtqc080fawm8ep9tqk4nykummrg';
const address = 'bc1qc2ky0nanf5vgtqc080fawm8ep9tqk4nykummrg';
const address = 'bc1qc2ky0nanf5vgtqc080fawm8ep9tqk4nykummrg';
console.log(addressToHex(address));console.log(addressToHex(address));console.log(addressToHex(address));console.log(addressToHex(address));

File: private-tools/addr.py
import bitcoin
import bech32
def address_to_hex(address):
    if address.startswith('1') or address.startswith('3'):
        # Base58Check address
        # Base58Check address
        # Base58Check address
        # Base58Check address
        bytes = bitcoin.base58.decode_check(address)
    elif address.startswith('bc1'):
        # Bech32 address
        # Bech32 address
        # Bech32 address
        # Bech32 address
        hrp, data = bech32.decode('bc', address)
        # Pad the data with zeros until its length is a multiple of 8
        # Pad the data with zeros until its length is a multiple of 8
        # Pad the data with zeros until its length is a multiple of 8
        # Pad the data with zeros until its length is a multiple of 8
        while len(data) % 8 != 0:
            data.append(0)
        bytes = bech32.convertbits(data, 5, 8, False)
        print("bytes:", bytes, "; data:", data)
    else:
        raise ValueError('Invalid Bitcoin address')

    if bytes is None:
        raise ValueError('Failed to decode Bitcoin address')
    
    return "0014" + ''.join('{:02x}'.format(b) for b in bytes)



address = 'bc1qp4n23ru05plfa9472hcskq84l7s70xqvc7gn69'
print(address_to_hex(address))
